pallets/capacity/src/lib.rs

```diff
@@ -513,24 +516,86 @@ impl<T: Config> Pallet<T> {
 		CapacityLedger::<T>::insert(target, capacity_details);
 	}
 
-	/// Decrease a staking account's active token and create an unlocking chunk to be thawed at some future block.
+	/// Decrease a staking account's active token.
 	fn decrease_active_staking_balance(
 		unstaker: &T::AccountId,
 		amount: BalanceOf<T>,
 	) -> Result<BalanceOf<T>, DispatchError> {
-		let mut staking_account =
-			Self::get_staking_account_for(unstaker).ok_or(Error::<T>::NotAStakingAccount)?;
-		ensure!(amount <= staking_account.active, Error::<T>::AmountToUnstakeExceedsAmountStaked);
+		StakingAccountLedger::<T>::try_mutate_exists(
```

Comment:
was there a reason we switch it to using a callback here?

Reply:
something along the lines of
```
let mut staking_account = Self::get_staking_account_for(unstaker).ok_or(Error::<T>::NotAStakingAccount)?;
ensure!(amount <= staking_account.active, Error::<T>::AmountToUnstakeExceedsAmountStaked);

let actual_unstaked_amount = staking_account.withdraw(amount)?;

if staking_account.active.is_zero() {
   StakingAccountLedger:: remove()
} else {
 StakingAccountLedger::set()
}
```

Reply:
I forgot about this one - this was here before we discussed moving away from `try_mutate` forms.